Upgrade Sentry, Requests, Boto3 to latest version

AmitPhulera opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->
When Dependabot raised a PR to update Sentry tests failed because a new deprecation warning was introduced in the code. This new deprecation was added by `urllib3` in their version `1.26.12`.

Our version of Boto3 was using this deprecated module and hence we were seeing the warning.

In the process of figuring out the cause, I came across https://github.com/urllib3/urllib3/issues/2680 which suggested us to get on a newer version of requests as well.

I was pairing with @gherceg yesterday and investigated this issue. On finding the root cause of deprecation we tried upgrading `urllib` to the latest version we came across a new deprecation warning which was coming from the version of elasticsearch that we are using.

So based on our findings I created this PR that
- Introduces a new command in make file to upgrade a specific dependency.
- Updates Sentry, boto3 and requests
- Adds the warning thrown by ES module to ignore list.

Review by commit 🐟 

<hr/>

## Information regarding boto3 update

We tried updating boto3 to the latest version which was 1.26.108 but we realised that tests started failing with 403. A stacktrace from one of the tests -


```

  ERROR: corehq.blobs.tests.test_migrate_compress:TestCompressMigration.test_compress_all
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/mnt/commcare-hq-ro/corehq/blobs/tests/test_migrate_compress.py", line 29, in setUp
      self.db.put(BytesIO(data), meta=meta)
    File "/mnt/commcare-hq-ro/corehq/blobs/s3db.py", line 70, in put
      s3_bucket = self._s3_bucket(create=True)
    File "/mnt/commcare-hq-ro/corehq/blobs/s3db.py", line 156, in _s3_bucket
      self.db.meta.client.head_bucket(Bucket=self.s3_bucket_name)
    File "/usr/local/lib/python3.9/site-packages/botocore/client.py", line 530, in _api_call
      return self._make_api_call(operation_name, kwargs)
    File "/usr/local/lib/python3.9/site-packages/botocore/client.py", line 960, in _make_api_call
      raise error_class(parsed_response, operation_name)
  botocore.exceptions.ClientError: An error occurred (403) when calling the HeadBucket operation: Forbidden

```

But this was only observed in the tests that were running on minio container and not on staging environment that was actually dealing with S3. 

So for the time being we are setting boto3 to a version that is making our tests happy. 

More info on the slack thread https://dimagi.slack.com/archives/C01G688ECF9/p1680878372067559

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Would be putting it out on staging for testing.
### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
Not sure if QA should be requested on this one as the upgrades were compatible with all the dependencies and have gone through changelog where nothing looks suspicious.

<!-- Please link to any past code changes that are coordinated with this migration -->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

AmitPhulera committed:
update makefile to support upgrading specific packages

AmitPhulera committed:
upgrade sentry

AmitPhulera committed:
upgrade requests

AmitPhulera committed:
ignore warning coming from es2 using deprecated code of urllib3

gherceg commented on requirements/Makefile:
```diff
@@ -17,3 +17,14 @@ upgrade-requirements:
+	pip-compile --upgrade-package $(package) -o test-requirements.txt test-requirements.in --allow-unsafe
+	pip-compile --upgrade-package $(package) -o dev-requirements.txt dev-requirements.in --allow-unsafe
+	pip-compile --upgrade-package $(package) -o docs-requirements.txt docs-requirements.in --allow-unsafe
+	python ../scripts/purge-platform-pkgs.py ./*requirements.txt
```
add newline

AmitPhulera replied:
[09891ca](https://github.com/dimagi/commcare-hq/pull/32820/commits/09891ca716a628a59c7d2e6e742b1b6b7e6d89d1)

AmitPhulera committed:
new line at EOF

millerdev commented on requirements/base-requirements.in:
```diff
@@ -1,7 +1,9 @@
 architect
 attrs>=21.4
-boto3
+# Updating to a higher version breaks tests running in minio more on https://github.com/dimagi/commcare-hq/pull/32820
+# Can be updated if the tests are resolved
+boto3==1.24.96
```
I'm not sure how aggressively pip-tools or dependabot will attempt to upgrade this library if it is not pinned? If they don't attempt to upgrade it frequently we could drop commit https://github.com/dimagi/commcare-hq/pull/32820/commits/8de1ee44ca1443b76ca520b5d991db8396044f40 and add this comment in the commit message of https://github.com/dimagi/commcare-hq/pull/32820/commits/b91c247a46def6afc7a1f7263290515460e718cb. That way upgrades are not strictly constrained, and this library may naturally upgrade to a newer version where the issue is fixed without us having to explicitly remove the pin.

AmitPhulera replied:
Your suggestion made so much sense in terms of not pinning boto3. I have updated the PR with the changes.

AmitPhulera committed:
pin boto to 1.24.96 in requirements only

Updating to a higher version breaks tests running in minio more on https://github.com/dimagi/commcare-hq/pull/32820
 Can be updated if the tests are resolved

AmitPhulera committed:
Merge remote-tracking branch 'origin/master' into ap/upgrade-urllib-and-related-deps

millerdev reviewed (approved):
I assume you're still working on the docs tests, but looks good so far.

millerdev committed:
Update urllib3 version for docs tests

urllib3>=1.26.13 causes an error in the docs tests:

Traceback (most recent call last):
  File ".../site-packages/sphinx/ext/autodoc/importer.py", line 60, in import_module
    return importlib.import_module(modname)
  ...
  File ".../corehq/__init__.py", line 43, in <module>
    assert _state._tls.current_app is None, "Current app already created"
AssertionError: Current app already created
